apps/mobile/src/features/balances/balances.tsx

```diff
@@ -1,42 +1,101 @@
+import { useRef, useState } from 'react';
+
 import {
   BitcoinBalance,
   BitcoinBalanceByAccount,
 } from '@/features/balances/bitcoin/bitcoin-balance';
 import { RunesBalance, RunesBalanceByAccount } from '@/features/balances/bitcoin/runes-balance';
 import { Sip10Balance, Sip10BalanceByAccount } from '@/features/balances/stacks/sip10-balance';
 import { StacksBalance, StacksBalanceByAccount } from '@/features/balances/stacks/stacks-balance';
-import { useRunesFlag } from '@/features/feature-flags';
+import { useRunesFlag, useTokenDetailsFlag } from '@/features/feature-flags';
+import { TokenSheet, TokenSheetData } from '@/features/token/token-sheet';
 import { ViewMode } from '@/shared/types';
 
-import { AccountId } from '@leather.io/models';
-import { Box } from '@leather.io/ui/native';
+import { AccountId, FungibleCryptoAsset, Money } from '@leather.io/models';
+import { Box, SheetRef } from '@leather.io/ui/native';
 
 export interface BalanceViewProps {
   mode: ViewMode;
+  onPress?: ({ asset, availableBalance, quoteBalance }: OnOpenTokenProps) => void;
+}
+
+export interface OnOpenTokenProps {
+  asset: FungibleCryptoAsset;
+  availableBalance: Money;
+  quoteBalance: Money;
 }
 
 export function AllAccountBalances({ mode }: BalanceViewProps) {
+  const [sheetData, setSheetData] = useState<TokenSheetData | null>(null);
   const runesFlag = useRunesFlag();
+  const tokenDetailsFlag = useTokenDetailsFlag();
+
+  const tokenSheetRef = useRef<SheetRef>(null);
+
+  function onOpenToken({ asset, availableBalance, quoteBalance }: OnOpenTokenProps) {
+    setSheetData({ asset, availableBalance, quoteBalance });
+    // analytics.track('token_sheet_opened', { source: 'action_bar' });
+    tokenSheetRef.current?.present();
+  }
+
+  const onPressToken = tokenDetailsFlag ? onOpenToken : undefined;
+
   return (
-    <Box flex={1} height="100%">
-      <BitcoinBalance />
-      <StacksBalance />
-      <Sip10Balance mode={mode} />
-      {runesFlag && <RunesBalance mode={mode} />}
-    </Box>
+    <>
+      <Box flex={1} height="100%">
+        <BitcoinBalance onPress={onPressToken} />
+        <StacksBalance onPress={onPressToken} />
+        <Sip10Balance mode={mode} onPress={onPressToken} />
+        {runesFlag && <RunesBalance mode={mode} />}
+      </Box>
+      <TokenSheet data={sheetData} sheetRef={tokenSheetRef} />
+    </>
   );
 }
 
 export function AccountBalances({ mode, fingerprint, accountIndex }: AccountId & BalanceViewProps) {
+  const [sheetData, setSheetData] = useState<TokenSheetData | null>(null);
   const runesFlag = useRunesFlag();
+
+  const tokenSheetRef = useRef<SheetRef>(null);
+  const tokenDetailsFlag = useTokenDetailsFlag();
+
+  function onOpenToken(data: TokenSheetData) {
+    setSheetData(data);
+    // analytics.track('token_sheet_opened', { source: 'action_bar' });
+    tokenSheetRef.current?.present();
+  }
+
+  const onPressToken = tokenDetailsFlag ? onOpenToken : undefined;
```

Comment:
I have enabled `tokenDetailsFlag` in `dev` but off in production in launch darkly. 

There is no way to trigger the `TokenDetails` sheet without entering here first